pkg/operation/botanist/component/nodelocaldns/nodelocaldns.go

```diff
@@ -594,15 +594,44 @@ func (c *nodeLocalDNS) containerArg() string {
 }
 
 func (c *nodeLocalDNS) forceTcpToClusterDNS() string {
-	if c.values.ForceTcpToClusterDNS {
+
+	fromSpec := true
+	if c.values.NodeLocalDNS != nil && c.values.NodeLocalDNS.ForceTCPToClusterDNS != nil {
+		fromSpec = *c.values.NodeLocalDNS.ForceTCPToClusterDNS
+	}
+
+	fromAnnotation := true
+	if annotationValue, err := strconv.ParseBool(c.values.ShootAnnotations[v1beta1constants.AnnotationNodeLocalDNSForceTcpToClusterDns]); err == nil {
```

Comment:
We should get rid of these annotations now that we have the fields in the `Shoot` API (not in this PR, but soon).